shared/src/commonMain/kotlin/com/bunbeauty/shared/data/repository/MenuProductRepository.kt

```diff
@@ -47,13 +52,30 @@ class MenuProductRepository(
     }
 
     override suspend fun getMenuProductByUuid(menuProductUuid: String): MenuProduct? {
-        return menuProductDao.getMenuProductByUuid(uuid = menuProductUuid).let {
-            if (it == null) return null
-            menuProductMapper.toMenuProduct(it)
-        }
+
+        return getCacheOrListData(
+            onApiRequest = networkConnector::getMenuProductList,
+            onLocalRequest = {
```

Comment:
Could you pls split this lambda into multiple functions to make it more readable?